README.md

```diff
@@ -1,2 +1,65 @@
-# message-queue-toolkit
-Useful utilities, interfaces and base classes for message queue handling
+# message-queue-toolkit ✉️
+Useful utilities, interfaces and base classes for message queue handling.
+
+## Overview
+
+This is an abstraction to switch between different queue systems without having to implement your own deserialization, error handling, etc. The library provides utilities, interfaces and base classes to build the support for any queue system you may need in your service and already implements support for the following:
+
+* AMQP (Advanced Message Queuing Protocol)
+* SQS (Simple Queue Service)
+* SNS (Simple Notification Service)
+
+## Basic Usage
+
+### Publishers
+
+Build a publisher to send messages to a queue or a topic in a messaging system by creating a class that:
```

Comment:
I would start this section with something along the lines of

`message-queue-toolkit provides base classes for implementing publishers for each of the supported protocol. They implement the following public methods:

constructor // explain parameters

init // explain that it needs to be invoked before publisher can be used

close // explain it needs to be invoked when stopping application

publish // explain that first parameter is the message payload, and second is protocol-dependent set of message parameters, you can also link to SQS/SNS/amqp-node documentation on their message parameters.
`